src/potdata/samplers.py

```diff
@@ -616,3 +616,244 @@ def plot2(self, separation1: int, separation2: int, show: bool = False,
 
         if show:
             plt.show()
+
+class KMeansStructureSampler(BaseStructureSampler):
+    """Sample structures using KMeans clustering.
+
+    Args:
+        kmeans_kwargs: Arguments to pass to `sklearn.cluster.KMeans`. You should
+            provide the `n_clusters` argument to specify the number of clusters.
+    """
+
+    DEFAULT_SOAP_KWARGS = {"r_cut": 5.0, "n_max": 8, "l_max": 5, "periodic": True}
+    DEFAULT_KMEANS_KWARGS = {"n_clusters": 8}
+
+    def __init__(
+        self,
+        soap_kwargs: dict = None,
+        species_to_select: list[str] | None = None,
+        pool_method: str = "concatenate",
+        pca_dim: int | None = None,
+        kmeans_kwargs: dict = None,
+        ratio: float = 1.0,
```

Comment:
`ratio` should be typed annotated as `ratio: int | float = 1`. 

You should add docs for all arguments below the `Args` above. Most of them are the same as in DBSCANSampler, so you can refer the user to that. But you definitely need to explain `ratio` - what it does when an integer is passed and what it does when a float is provided.

Reply:
Done, thanks